crates/common/logged_command/src/logged_command.rs

```diff
@@ -148,23 +148,40 @@ impl std::fmt::Display for LoggedCommand {
 }
 
 impl LoggedCommand {
-    pub fn new(program: impl AsRef<OsStr>) -> LoggedCommand {
+    pub fn new(program: impl AsRef<OsStr>) -> Result<LoggedCommand, std::io::Error> {
```

Comment:
Very minor: I think `try_new()` is a better name as the function now returns a `Result`.